Mt7921e devel

objelf opened the pull request:
Let's continue work on pending patches.

I will check the issue you spotted on mt7663 with "mt76: mt7921: fix UC entry is being overwritten". 
and need your help on review and enhance the other patches

LorenzoBianconi commented on drivers/net/wireless/mediatek/mt76/mt76_connac_mcu.c:
```diff
@@ -936,6 +936,7 @@ void mt76_connac_mcu_wtbl_ba_tlv(struct mt76_dev *dev, struct sk_buff *skb,
 		ba->rst_ba_tid = params->tid;
 		ba->rst_ba_sel = RST_BA_MAC_TID_MATCH;
 		ba->rst_ba_sb = 1;
+		ba->ba_winsize = enable ? cpu_to_le16(params->buf_size) : 0;
```
@objelf since mt76_connac_mcu_wtbl_ba_tlv() is used by mt7615 and mt7663 I guess we should move this inside the if (is_mt7921()) check. I will fix it applying the patch

LorenzoBianconi commented on drivers/net/wireless/mediatek/mt76/mt7921/main.c:
```diff
@@ -295,12 +247,7 @@ static int mt7921_add_interface(struct ieee80211_hw *hw,
-		goto out;
-	}
-	mvif->mt76.omac_idx = idx;
+	mvif->mt76.omac_idx = mvif->mt76.idx;
```
@objelf have you checked the possible value for omac_idx according to the irc discussion?

moore-bros committed:
mt76: connac: add mt7921_mcu_sta_update support

Add mt76_connac_mcu_update_sta_cmd support that is the prerequisite patch for
we can update the station record with the current state the station has into
the offload firmware at runtime.

That is the prerequisite patch to fix sta_state incorrect
implementation.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>

moore-bros committed:
mt76: mt7921: fix sta_state incorrect implementation

When .sta_state is implemented, mac80211 assumes that the station entry is
usable after the NOTEXIST->NONE transition

So we should create the sta entry as early as possible in order that
mac80211 pass assoc/auth frames to mt76 with the newly created sta entry,
and add .sta_assoc to mt76 core to refresh the sta entry again when sta is
being associated.

Fixes: c0e93934635a ("mt76: mt7921: enable deep sleep at runtime")
Signed-off-by: Sean Wang <sean.wang@mediatek.com>

moore-bros committed:
mt76: mt7921: fix cannot fw own sometime

Fix the driver cannot ensure in fw own state when the device is idle for
a while.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>

moore-bros committed:
mt76: connac: add mt76_connac_mcu_update_sta_cmd support

Add mt76_connac_mcu_update_sta_cmd support that is the prerequisite patch for
we can update the station record with the current state the station has into
the offload firmware at runtime.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>
